setup.py

```diff
@@ -22,10 +22,140 @@
 #
 
 from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext
+import sysconfig
+import platform
 
 with open("README.md", "r") as file_long_description:
     long_description = file_long_description.read()
 
+BUILD_DIR = "build"
+machine = platform.machine().lower()  # TODO: this is unreliable, need something better to detect target architecture
+is_armv7 = (machine.startswith("armv7"))  # TODO: can armv8_ still be AArch32?
+is_arm = (is_armv7 or machine.startswith("arm") or machine.startswith("aarch64"))
+is_x86 = (machine in ["i386", "i686", "x86", "x86_64", "x64", "amd64"])
+
+
+
+class sabyenc_build(build_ext):
```

Comment:
Nice, you already found it! I was just about to write that this seems indeed be the way to implement any custom build work.